memory/src/main/java/org/opensearch/ml/memory/index/InteractionsIndex.java

```diff
@@ -362,26 +455,26 @@ public void searchInteractions(String conversationId, SearchRequest request, Act
             if (access) {
                 try (ThreadContext.StoredContext threadContext = client.threadPool().getThreadContext().stashContext()) {
                     ActionListener<SearchResponse> internalListener = ActionListener.runBefore(listener, () -> threadContext.restore());
-                    request.indices(INTERACTIONS_INDEX_NAME);
+                    request.indices(indexName);
                     QueryBuilder originalQuery = request.source().query();
                     BoolQueryBuilder newQuery = new BoolQueryBuilder();
                     newQuery.must(originalQuery);
                     newQuery.must(new TermQueryBuilder(ConversationalIndexConstants.INTERACTIONS_CONVERSATION_ID_FIELD, conversationId));
                     request.source().query(newQuery);
-                    client
-                        .admin()
-                        .indices()
-                        .refresh(Requests.refreshRequest(INTERACTIONS_INDEX_NAME), ActionListener.wrap(refreshResponse -> {
-                            client.search(request, internalListener);
-                        }, e -> {
-                            log.error("Failed to refresh interactions index during search interactions ", e);
-                            internalListener.onFailure(e);
-                        }));
+                    client.admin().indices().refresh(Requests.refreshRequest(indexName), ActionListener.wrap(refreshResponse -> {
```

Comment:
Same here, why refresh before searching ? How about change to refresh for write request (create/update)?